test/unit/scripts/resolver.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+module.exports = ( path, options ) => {
+	// Call the defaultResolver, so we leverage its cache, error handling, etc.
+	return options.defaultResolver( path, {
+		...options,
+		// Use packageFilter to process parsed `package.json` before the resolution (see https://www.npmjs.com/package/resolve#resolveid-opts-cb)
+		packageFilter: ( pkg ) => {
+			// This is a workaround for https://github.com/uuidjs/uuid/pull/616
+			//
+			// jest-environment-jsdom 28+ tries to use browser exports instead of default exports,
+			// but uuid only offers an ESM browser export and not a CommonJS one. Jest does not yet
+			// support ESM modules natively, so this causes a Jest error related to trying to parse
+			// "export" syntax.
```

Comment:
note that uuid v9 does expose CJS entrypoint for browser env.

You might still need this custom resolver, just pointing out that the comment is somewhat stale 🙂

Reply:
Thanks, I see the browser export was fixed in https://github.com/uuidjs/uuid/pull/642. I'll change the comment to mention "uuid v8" explicitly. After we upgrade to uuid v9, we can remove this workaround. I'd rather do the v9 upgrade in a separate PR, I'm not even sure if it will be feasible, as there are also 3rd party libraries which can use old version of `uuid`.